File: algorithms/2_medium/A/03021_alice_and_bob_playing_flower_game/solution.c
long long flowerGame(const int n, const int m)
{
    const long long even_cnt_n = n / 2;
    const long long odd_cnt_n = n - even_cnt_n;

    const long long even_cnt_m = m / 2;
    const long long odd_cnt_m = m - even_cnt_m;

    return even_cnt_n * odd_cnt_m + odd_cnt_n * even_cnt_m;
}

File: algorithms/2_medium/A/03021_alice_and_bob_playing_flower_game/solution.rs
impl Solution {
    pub fn flower_game(n: i32, m: i32) -> i64 {
        let even_cnt_n = (n as i64) / 2;
        let odd_cnt_n = (n as i64) - even_cnt_n;

        let even_cnt_m = (m as i64) / 2;
        let odd_cnt_m = (m as i64) - even_cnt_m;

        even_cnt_n * odd_cnt_m + odd_cnt_n * even_cnt_m
    }
}
